eventsourcing/src/main/java/org/axonframework/eventsourcing/AbstractSnapshotter.java

```diff
@@ -92,13 +97,28 @@ private void doScheduleSnapshot(Class<?> aggregateType, String aggregateIdentifi
             }
         }
         if (snapshotsInProgress.add(typeAndId)) {
+            String rootTraceName = String.format("%s.createSnapshot(%s)",
```

Comment:
It would be nice to delay the call to format to within the createRootTrace method. Formatting strings is relatively expensive since it needs to be done for every invocation. When no tracing is configured, this call is (presumably) still performed. It could be that the compiler optimizes this, but that's a pure guess at this point.
I am thinking something along the lines that the loggers also use. Or alternatively, allow message to be a `Supplier<String> messageSupplier`.

Reply:
All methods now take a `Supplier<String> spanNameSupplier`. I converted all `String.format` calls to string concatenation as well.